rules/utils/avoid-capture.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+const resolveVariableName = require('./resolve-variable-name');
+
+const indexifyName = (name, index) => index === 1 ? name : name + index;
+
+const someScopeHasVariableName = (name, scopes) => scopes.some(scope => resolveVariableName(name, scope));
+
+module.exports = (name, scopes) => {
```

Comment:
Can you add doc comment on what this does and what problem it solves? For posterity.